### DISCUSSION TITLE
Add members to AdministrativeUnits

### DISCUSSION THREAD

**Discussion by summonmayank** [sreeise/graph-rs-sdk#372] [Q&A]
Hello,

I am trying to add members to administrative unit using graph-rs-sdk but I don't see a method for it. It has the method to list members "list_members" or "get_members". According to the docs, there should be a method to add members ? Am I missing something here ?
Here is the request mentioned in the docs:

```
POST https://graph.microsoft.com/v1.0/directory/administrativeUnits/{id}/members/$ref
Content-type: application/json

{
  "@odata.id":"https://graph.microsoft.com/v1.0/groups/{id}"
}

```

I think there is no API implemented for adding members to administrative units ?

Thanks

**Comment by sreeise** [comment]
Hey, this is most likely due to either 1 or 2 things:

1. The requests available are generated from the OpenApi config that Micorosoft provides and the config did not include a request for adding members.
2. There was some issue parsing the OpenApi config and that request didnt get included. 

Sometimes not all requests that can be used are included in their the OpenApi config and they typically get updated later on. That is probably what happened here.  From time to time I will go back and regenerate the requests from an updated OpenApi config and it looks like it needs to be done here with the `/directory` endpoint. I am checking now to see if that specific request would be included and ill get back to you.

**Reply by sreeise** [reply]
Looks like the request to add members to the administrative unit have been included in the OpenApi config. I created a ticket, https://github.com/sreeise/graph-rs/issues/375, to get those added. It might take a little bit of time because it looks like they have added a lot of newer api requests to it as well and its not always a 1 to 1 mapping when generating the methods for Rust. So this just means I usually have to go in and add some settings to make sure those get generated correctly.

**Reply by summonmayank** [reply]
Great ! Thanks for the update, Sean !

**Reply by sreeise** [reply]
@summonmayank Just merged changes into master for Directory. You should be able to do call the method you mentioned now:

```rust
   let create_member_json = serde_json::({});

    client
        .v1()
        .directory()
        .administrative_unit("administrative-unit-id")
        .members()
        .create_ref_members(&create_member_json);
```

**Reply by summonmayank** [reply]
Great ! Thanks !
